Dockerfile.builder

```diff
@@ -16,31 +13,47 @@ RUN set -ex \
     B9AE9905FFD7803F25714661B63B535A4C206CA9 \
     C4F0DFFF4E8C1A8236409D08E73BC641CC11F4C8 \
   ; do \
-    gpg --import "/gpg/${key}.gpg" ; \
+    gpg --keyserver ha.pool.sks-keyservers.net --recv-keys "$key" || \
+    gpg --keyserver pgp.mit.edu --recv-keys "$key" || \
+    gpg --keyserver keyserver.pgp.com --recv-keys "$key" ; \
   done
 
 #ENV NPM_CONFIG_LOGLEVEL info
 ENV NODE_VERSION 6.9.2
 
-RUN yum install -y bzip2 fontconfig java-1.8.0-openjdk nmap-ncat psmisc git \
-  && curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.xz" \
-  && curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/SHASUMS256.txt.asc" \
-  && gpg --batch --decrypt --output SHASUMS256.txt SHASUMS256.txt.asc \
-  && grep " node-v$NODE_VERSION-linux-x64.tar.xz\$" SHASUMS256.txt | sha256sum -c - \
-  && tar -xJf "node-v$NODE_VERSION-linux-x64.tar.xz" -C /usr/local --strip-components=1 \
-  && rm "node-v$NODE_VERSION-linux-x64.tar.xz" SHASUMS256.txt.asc SHASUMS256.txt \
-  && ln -s /usr/local/bin/node /usr/local/bin/nodejs \
-  && yum clean all
+USER root
+RUN yum -y update && \
+    yum install -y bzip2 fontconfig tar java-1.8.0-openjdk nmap-ncat psmisc gtk3 git \
+      python-setuptools xorg-x11-xauth wget unzip which xorg-x11-server-Xvfb \
+      xfonts-100dpi xorg-x11-fonts-75dpi xfonts-scalable xfonts-cyrillic \
+      ipa-gothic-fonts xorg-x11-utils xorg-x11-fonts-Type1 xorg-x11-fonts-misc \
+      GConf2 wget libXfont wget && \
+      yum -y clean all
+RUN yum install -y firefox
+
+RUN wget "http://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.gz" \
+  && tar -xvf "node-v$NODE_VERSION-linux-x64.tar.gz" -C /usr/local --strip-components=1 \
+  && rm "node-v$NODE_VERSION-linux-x64.tar.gz" \
+  && ln -s /usr/local/bin/node /usr/local/bin/nodejs
+
+RUN  wget https://github.com/mozilla/geckodriver/releases/download/v0.14.0/geckodriver-v0.14.0-linux64.tar.gz && \
+  tar -xvf geckodriver-v0.14.0-linux64.tar.gz && \
+  chmod +x geckodriver && \
+  mv geckodriver /usr/bin
+
+RUN npm install -g jasmine-node karma-firefox-launcher protractor
 
+ENV DISPLAY=:99
 ENV FABRIC8_USER_NAME=fabric8
 
 RUN useradd --user-group --create-home --shell /bin/false ${FABRIC8_USER_NAME}
 
 ENV HOME=/home/${FABRIC8_USER_NAME}
-
 ENV WORKSPACE=$HOME/fabric8-planner
 RUN mkdir $WORKSPACE
 
+RUN chmod u+s /usr/bin/Xvfb && chown fabric8 /usr/bin/Xvfb
```

Comment:
1. What is this line good for?
2. Is shouldn't be 'fabric8' string replaced with `${FABRIC8_USER_NAME}`?